rsocket-core/src/main/java/io/rsocket/RSocketFactory.java

```diff
@@ -109,6 +111,18 @@ static ServerRSocketFactory receive() {
         private String dataMineType = "application/binary";
         private String metadataMimeType = "application/binary";
 
+        private @Nullable RSocketStats stats;
+
+        public ClientRSocketFactory stats(RSocketStats stats) {
+            this.stats = stats;
+
+            if (stats != null) {
+                stats.socketCreated();
```

Comment:
Is this possibly too early? e.g. can the builder be left in a dry state, and triggered later?